### PR TITLE
Add Wormhole Gateway initial assets

### PR DESCRIPTION
**Pull request by nik-suri**

### CODE REVIEW

**Comment by omniwired** [comment]
hi @nik-suri please check why the validation is failing using:
> yarn validate cosmos/{your file.json}

File: cosmos/wormchain.json [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+      "coinMinimalDenom": "factory/wormhole14ejqjyq8um4p3xfqj74yld5waqljf88fz25yxnma0cngspxe3les00fpjx/GGh9Ufn1SeDGrhzEkMyRKt5568VbbxZK2yvWNsd6PbXt",
+      "coinDecimals": 6,
+      "coinGeckoId": "usd-coin",
+      "coinImageUrl": "https://raw.githubusercontent.com/chainapsis/keplr-chain-registry/main/images/_non-cosmos/usdc.png"
+    }
+  ],
+  "feeCurrencies": [
+    {
+      "coinDenom": "TEST",
+      "coinMinimalDenom": "utest",
```
Comment (omniwired) [inline cosmos/wormchain.json:111]: maybe it should be `worm` ?

**Comment by HeesungB** [comment]
@nik-suri @omniwired Could you check your node?
```
Failed to connect websocket to /websocket rpc
```

File: cosmos/wormchain.json (hunk 1 above)
Comment (nik-suri) [reply cosmos/wormchain.json:111]: no, `test` is correct

**Comment by omniwired** [comment]
I see, I was unable to bridge (ibc) from osmosis back to wormhole, is this lack of websocket related in any way?

**Comment by nik-suri** [comment]
@omniwired how are you attempting to bridge back? Simple ICS20 transfers from Osmosis to Gateway won't work. The transfers need to have a specially formatted memo. See https://docs.wormhole.com/wormhole/explore-wormhole/gateway for details. 

The only frontends that currently support properly bridging in/out of Osmosis via Wormhole Gateway are [Wormhole Connect](https://wormhole.com/connect/) and [Portal](https://portalbridge.com/cosmos/) (which uses Wormhole Connect under the hood).

**Comment by omniwired** [comment]
@HeesungB is it possible to merge without a working endpoint?
@nik-suri what's the best to reach you for the IBC issue? I'm on the wormhole discord (omniwired) and in telegram (omniwired)

**Comment by nik-suri** [comment]
Thanks @omniwired - I will reach out to you

**Comment by HeesungB** [comment]
@omniwired Because Keplr uses websocket, if your endpoint doesn't support websocket, you can't use it with Keplr.

**Comment by HeesungB** [comment]
@nik-suri @omniwired 
If you post your chain information here, it may be unintentionally available to users.
If that happens, they may have trouble using `Keplr`.

It might be a good idea to add a websocket.

**Comment by nik-suri** [comment]
@HeesungB got it. How can we make sure that Wormhole assets that have been bridged to Osmosis are properly recognized in Keplr without posting the chain information as well?

For example, see the assets recognized on Mintscan here - https://www.mintscan.io/osmosis/accounts/osmo108jrl3qtt5aaqy5gcau6w8v4pgl22tumdzqdl4

Leap Wallet also properly recognizes the assets.

**Comment by nik-suri** [comment]
> @HeesungB got it. How can we make sure that Wormhole assets that have been bridged to Osmosis are properly recognized in Keplr without posting the chain information as well?
> 
> For example, see the assets recognized on Mintscan here - https://www.mintscan.io/osmosis/accounts/osmo108jrl3qtt5aaqy5gcau6w8v4pgl22tumdzqdl4
> 
> Leap Wallet also properly recognizes the assets on Osmosis.

Hey @HeesungB following up here - How can we make sure that Wormhole assets that have been bridged to Osmosis are properly recognized in Keplr without posting the chain information as well?

**Comment by dogemos** [comment]
This is not feasible right now. Keplr operates on the dynamic fetching of IBC token information, where token information is received from the original chain of the token (in this case, `wormchain`).

This requires the chain to be visible, and enabled on the user's wallet. Showing the token, but not showing the chain will require additional engineering work. We may tackle this in the future, but given our existing commitment on resources, we can't guarantee when/if this will be available.

**Comment by nik-suri** [comment]
@dogemos @HeesungB could you please re-review this PR?
Gateway now has a public RPC provider that exposes the websocket URL. Would love to get this PR merged and have support in Keplr for these assets!

cc @omniwired

**Comment by HeesungB** [comment]
@nik-suri Please add an image like the link example below.

{chiainIdentifier}/{factory token minimal denom}/{minimal denom}.png

https://github.com/chainapsis/keplr-chain-registry/tree/main/images/osmosis/factory/osmo14klwqgkmackvx2tqa0trtg69dmy0nrg4ntq4gjgw2za4734r5seqjqm4gm

**Comment by nik-suri** [comment]
@HeesungB to make sure I understand, I should organize each denom like the Osmosis factory ones above, instead of putting them in a `_non-cosmos` directory?

**Comment by omniwired** [comment]
This one will be even more important after yesterday integration by Osmosis.
Can we revisit this?

**Comment by HeesungB** [comment]
@omniwired Could you resize image to 256x256px

**Comment by HeesungB** [comment]
@omniwired I can try to see if you can handle this by making the validation pass.
Rebase the main branch and start

**Pull request merged**